creme/compat/creme_to_sklearn.py

```diff
@@ -99,42 +99,73 @@ class Creme2SKLRegressor(Creme2SKLBase, sklearn_base.RegressorMixin):
     """
 
     def __init__(self, estimator: base.Regressor):
+
+        # Check the estimator is a Regressor
+        if not isinstance(estimator, base.Regressor):
+            raise ValueError('estimator is not a Regressor')
+
         self.estimator = estimator
 
+    def _partial_fit(self, X, y):
+
+        # Check the inputs
+        X, y = utils.check_X_y(X, y, **SKLEARN_INPUT_X_PARAMS, **SKLEARN_INPUT_Y_PARAMS)
+
+        # Store the number of features so that future inputs can be checked
+        if hasattr(self, 'n_features_in_') and X.shape[1] != self.n_features_in_:
+            raise ValueError(f'Expected {self.n_features_in_} features, got {X.shape[1]}')
+        self.n_features_in_ = X.shape[1]
+
+        # scikit-learn's convention is that fit shouldn't mutate the input parameters; we have to
+        # deep copy the provided estimator in order to respect this convention
+        if not hasattr(self, 'instance_'):
+            self.instance_ = copy.deepcopy(self.estimator)
+
+        # Call fit_one for each observation
+        for x, yi in STREAM_METHODS[type(X)](X, y):
+            self.instance_.fit_one(x, yi)
+
+        return self
+
     def fit(self, X, y):
         """Fits to an entire dataset contained in memory.
 
         Parameters:
-            X (array-like of shape (n_samples, n_features))
-            y (array-like of shape n_samples)
+            X: array-like of shape (n_samples, n_features).
+            y: array-like of shape n_samples.
 
         Returns:
             self
 
         """
 
-        # Check the estimator is a Regressor
-        if not isinstance(self.estimator, base.Regressor):
-            raise ValueError('estimator is not a Regressor')
+        # Reset the state if already fitted
+        for attr in ('instance_', 'n_features_in_'):
+            self.__dict__.pop(attr, None)
 
-        # Check the inputs
-        X, y = utils.check_X_y(X, y, **SKLEARN_INPUT_X_PARAMS, **SKLEARN_INPUT_Y_PARAMS)
+        # Fit with one pass of the dataset
+        return self._partial_fit(X, y)
 
-        # scikit-learn's convention is that fit shouldn't mutate the input parameters; we have to
-        # deep copy the provided estimator in order to respect this convention
-        self.instance_ = copy.deepcopy(self.estimator)
+    def partial_fit(self, X, y, classes=None):
+        """Fits incrementally on a portion of a dataset.
 
-        # Call fit_one for each observation
-        for x, yi in STREAM_METHODS[type(X)](X, y):
-            self.instance_.fit_one(x, yi)
+        Parameters:
+            X: array-like of shape (n_samples, n_features).
+            y: array-like of shape n_samples.
 
-        return self
+        Returns:
+            self
+
+        """
+
+        # Fit with one pass of the dataset portion
+        return self._partial_fit(X, y)
```

Comment:
Is there any reason why need `_partial_fit`? Can't we just have `partial_fit`?

Reply:
It's similar to how it's [done](https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/linear_model/_stochastic_gradient.py#L449) it `sklearn`. `_partial_fit` contains the common denominator to `fit` and `partial_fit`. The current `Creme2SKL*` are simple enough that there is nothing else to put in `partial_fit` but theoretically there could be some specific checks like [that](https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/linear_model/_stochastic_gradient.py#L680). Seems a fair design although I don't love the extra code either. I can put all in `partial_fit` if you prefer, just let me know.